src/commcare_cloud/ansible/roles/elasticsearch/defaults/main.yml

```diff
@@ -1,8 +1,8 @@
 ---
 elasticsearch_version: 2.4.6
 # https://www.elastic.co/guide/en/elasticsearch/guide/current/heap-sizing.html#_give_half_your_memory_to_lucene
-# Either half the machines RAM or 30 GB. Never want to go over 30 GB
-elasticsearch_memory: "{{ [ansible_memory_mb.real.total // 2, 30000] | min }}m"
+# Either half the machines RAM or 24 GB. If changing in the future, never want to go over 30 GB
+elasticsearch_memory: "{{ [ansible_memory_mb.real.total // 2, 24576] | min }}m"
```

Comment:
Why 24GB? Is that custom tuning for our production cluster, or is it a generic limit that makes sense on all clusters? If it's specific to production, maybe it makes sense to make this setting parameterized per environment?

Reply:
I'll let @calellowitz confirm, but my understanding of the rationale is basically (1) our production instance is the biggest ES cluster of any CommCare instance so if it's okay for us to use 24GB heap, it's probably okay for anyone (this is a bit handwavy about different possible configurations, etc, but most third party instances are monoliths where presumably you'd already want ES to take up less space) and (2) if we're running into this GC issue, then other people who got to somewhere near our size and usage might also run into it, so they'd benefit from lowering the heap size for the same reason as we do. I don't feel super strongly about it, but I think that rationale makes reasonable sense.

Reply:
Yeah, basically what Danny said. That we believe this to be the largest cluster by far, and that this is an easy mistake to make but a hard one to find, so if we can use defaults to prevent people from over committing, that might be helpful (and this whole setting can still be set per environment in the public yaml  files). I don't feel particularly strongly though, so if you do, happy to change it.

Reply:
Thanks. Sounds fine if it can be overridden per env as needed.